js_modules/dagster-ui/packages/ui-core/src/app/Permissions.tsx

```diff
@@ -111,53 +113,67 @@ type PermissionDisabledReasons = Record<keyof PermissionsMap, string>;
 export type PermissionsState = {
   permissions: PermissionBooleans;
   disabledReasons: PermissionDisabledReasons;
-  loading: boolean;
 };
 
-type PermissionsContextType = {
+export type PermissionsResult = {
+  error?: ApolloError;
   unscopedPermissions: PermissionsMap;
   locationPermissions: Record<string, PermissionsMap>;
-  loading: boolean;
-  // Raw unscoped permission data, for Cloud extraction
   rawUnscopedData: PermissionFragment[];
 };
 
+type PermissionsContextType = ReturnType<typeof wrapPromise<PermissionsResult>>;
```

Comment:
Would be nice for your wrapPromise util to expose a type helper that makes this `Wrapped<PermissionsResult>`,  I expect we'll need this in a lot of places